Automattic/newspack-blocks
# fix: check for purchaseable products merged from trunk

### All Submissions:


### Changes proposed in this Pull Request:

[This PR](https://github.com/Automattic/newspack-blocks/pull/1916) changed code that was very different between trunk and epic/ras-acc, so I made this one to merge trunk into epic/ras-acc early, and to remove the NYP check we don't need in RAS-ACC.

See 1207817176293825-as-1208327956169322

### How to test the changes in this Pull Request:

Follow the steps to test from https://github.com/Automattic/newspack-blocks/pull/1916, minus the NYP comment -- NYP product should be allowed to be added to the Checkout Button block and the block should show on the front-end because readers can set a purchase price from the checkout.

### Other information:
